### PR TITLE
[ARTS-355] - Limit public/internet access to API services

### PR DESCRIPTION
**Pull request by eladiw**
## Description
Limit public/internet access to API services.
ALL SQL, KV  and web apps (except gateway) are now inside a vnet and communicate using vnet integration + private endpoints.
the gateway has vnet integration for outbound connections and remains publicly available.

### Changes
- [ARTS- 355] - Limit public/internet access to API services.

### Checklist:

- [x] Branch name follows convention (feature/arts-###-short-name OR fix/arts-###-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

### CODE REVIEW

File: terraform/trial_rg/modules/privateendpoint/main.tf [hunk 1]
```diff
@@ -21,14 +6,14 @@ resource "azurerm_private_endpoint" "private_endpoint" {
 
   private_dns_zone_group {
     name                 = "${var.application}privatednszonegroup"
```
Comment (tamirkamara) [inline terraform/trial_rg/modules/privateendpoint/main.tf]: can't this name be separated with "-"?

File: terraform/trial_rg/modules/privateendpoint/main.tf [hunk 2]
```diff
@@ -42,7 +27,7 @@ data "azurerm_private_endpoint_connection" "endpoint-connection" {
 # Create a Resource's Private DNS A Record
 resource "azurerm_private_dns_a_record" "endpoint-dns-a-record" {
   name                = lower("${var.application}-dns-record")
-  zone_name           = azurerm_private_dns_zone.endpoint-dns-private-zone.name
+  zone_name           = var.dns_zone_name
   resource_group_name = var.rg_name
   ttl                 = 300
   records             = [data.azurerm_private_endpoint_connection.endpoint-connection.private_service_connection.0.private_ip_address]
```
Comment (tamirkamara) [inline terraform/trial_rg/modules/privateendpoint/main.tf]: Looking at the private dns zone, this looks odd - you have 2 A records: this one you create here, and another one with a "nicer" name that I guess is created automatically, both are pointing to the same IP address. I don't see you reference/use this one anywhere, why do we need it?
This comment is also valid for the sql server (which, btw, don't follow the same name pattern for this resource), but oddly not for the keyvault...

File: terraform/trial_rg/modules/genericservice/main.tf [hunk 3]
```diff
@@ -29,4 +29,23 @@ resource "azurerm_app_service" "generic_service" {
   }
 
   app_settings = var.settings
+}
+
+# count = 0, if this is the gateway and no private endpoint is needed
+module "private_endpoint" {
+  count            = var.no_private_endpoint == true ? 0 : 1
```
Comment (tamirkamara) [inline terraform/trial_rg/modules/genericservice/main.tf]: perhaps this should be switched around to say enable_private_endpoint ?

File: terraform/trial_rg/modules/privateendpoint/variables.tf [hunk 4]
```diff
@@ -58,19 +53,23 @@ variable "vnet_id" {
...
-  type        = bool
-  description = "is related to kv."
-  default     = false
-}
-
-variable "sql" {
-  type        = bool
-  description = "is related to sql."
-  default     = false
+variable "subresource_name" {
```
Comment (tamirkamara) [inline terraform/trial_rg/modules/privateendpoint/variables.tf:56]: Not about this specifically, but looks like some of the vars aren't used...

**Review by tamirkamara — CHANGES_REQUESTED** [review]

**Review by mdp0 — DISMISSED** [review]

File: terraform/trial_rg/modules/privateendpoint/main.tf (hunk 2 above)
Comment (eladiw) [reply terraform/trial_rg/modules/privateendpoint/main.tf]: fixed

File: terraform/trial_rg/modules/genericservice/main.tf (hunk 3 above)
Comment (eladiw) [reply terraform/trial_rg/modules/genericservice/main.tf]: fixed

File: terraform/trial_rg/modules/privateendpoint/variables.tf (hunk 4 above)
Comment (eladiw) [reply terraform/trial_rg/modules/privateendpoint/variables.tf:56]: talked f2f. on the next PR, I will do some refactoring which will include fixing this item

**Review by tamirkamara — APPROVED** [review]

**Review by mdp0 — APPROVED** [review]

**Pull request merged**